azblob/zt_put_blob_from_url_test.go

```diff
@@ -0,0 +1,246 @@
+	bsu := getBSU()
+	credential, err := getGenericCredential("")
+	if err != nil {
+		c.Fatal("Invalid credential")
+	}
+	container, _ := createNewContainer(c, bsu)
+
+	testSize := size * 1024 * 1024 // 1MB
+	r, sourceData := getRandomDataAndReader(testSize)
+	sourceDataMD5Value := md5.Sum(sourceData)
+	srcBlob := container.NewBlockBlobURL(generateBlobName())
+	destBlob := container.NewBlockBlobURL(generateBlobName())
+
+	return container, credential, r, sourceData, sourceDataMD5Value, srcBlob, destBlob
+}
+
+func (s *aztestsSuite) TestPutBlockBlobFromURLWithTags(c *chk.C) {
+	container, credential, r, sourceData, sourceDataMD5Value, srcBlob, destBlob := CreateBlockBlobsForTesting(c, 1)
+	defer delContainer(c, container)
+
+	blobTagsMap := BlobTagsMap{
+		"Go":         "CPlusPlus",
+		"Python":     "CSharp",
+		"Javascript": "Android",
+	}
+
+	uploadSrcResp, err := srcBlob.Upload(ctx, r, BlobHTTPHeaders{}, Metadata{}, BlobAccessConditions{}, DefaultAccessTier, blobTagsMap, ClientProvidedKeyOptions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(uploadSrcResp.Response().StatusCode, chk.Equals, 201)
+
+	// Get source blob URL with SAS for StageFromURL.
+	srcBlobParts := NewBlobURLParts(srcBlob.URL())
+
+	srcBlobParts.SAS, err = BlobSASSignatureValues{
+		Protocol:      SASProtocolHTTPS,                     // Users MUST use HTTPS (not HTTP)
+		ExpiryTime:    time.Now().UTC().Add(48 * time.Hour), // 48-hours before expiration
+		ContainerName: srcBlobParts.ContainerName,
+		BlobName:      srcBlobParts.BlobName,
+		Permissions:   BlobSASPermissions{Read: true}.String(),
+	}.NewSASQueryParameters(credential)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	srcBlobURLWithSAS := srcBlobParts.URL()
+
+	// Invoke put blob from URL
+	resp, err := destBlob.PutBlobFromURL(ctx, BlobHTTPHeaders{}, srcBlobURLWithSAS, Metadata{"foo": "bar"}, ModifiedAccessConditions{}, BlobAccessConditions{}, sourceDataMD5Value[:], sourceDataMD5Value[:], DefaultAccessTier, blobTagsMap, ClientProvidedKeyOptions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(resp.Response().StatusCode, chk.Equals, 201)
+	c.Assert(resp.ETag(), chk.Not(chk.Equals), "")
+	c.Assert(resp.RequestID(), chk.Not(chk.Equals), "")
+	c.Assert(resp.Version(), chk.Not(chk.Equals), "")
+	c.Assert(resp.Date().IsZero(), chk.Equals, false)
+	c.Assert(resp.ContentMD5(), chk.DeepEquals, sourceDataMD5Value[:])
+
+	// Check data integrity through downloading.
+	downloadResp, err := destBlob.BlobURL.Download(ctx, 0, CountToEnd, BlobAccessConditions{}, false, ClientProvidedKeyOptions{})
+	c.Assert(err, chk.IsNil)
+	destData, err := ioutil.ReadAll(downloadResp.Body(RetryReaderOptions{}))
+	c.Assert(err, chk.IsNil)
+	c.Assert(destData, chk.DeepEquals, sourceData)
+	c.Assert(len(downloadResp.NewMetadata()), chk.Equals, 1)
+	c.Assert(downloadResp.r.rawResponse.Header.Get("x-ms-meta-foo"), chk.Equals, "bar")
```

Comment:
>c.Assert(downloadResp.r.rawResponse.Header.Get("x-ms-meta-foo"), chk.Equals, "bar") [](start = 1, length = 83)

You can check metadata in a much easier way, no need to go through the raw response. Please refer to the other tests.